### DISCUSSION TITLE
Can't chosse Entity "AC in" in Energy Dashboard

### DISCUSSION THREAD

**Discussion by csc-arnstadt** [vwt12eh8/hassio-ecoflow#45] [Q&A]
First: Great Job!!!! I love this Integration <3

One little thing I have. I can't choose Entity "AC in" in **Energy Dashboard**. I would prefer to have all the inputs and outputs to choose from dropdown. Is that possible? Home Assistant help say this ->

"You’re configuring a statistic but you couldn’t find your source in the dropdown? That’s caused by a bug in the integration providing the entity. Integrations need to configure their entities correctly so Home Assistant knows that we need to track statistics for it and how.
Open an issue with the author of the integration and link them to https://developers.home-assistant.io/docs/core/entity/sensor#long-term-statistics."

Thanx and best regads.
Rainer Nowak

**Comment by vwt12eh8** [comment]
The unit for the AC input entity is "W", so to add it to the energy dashboard you first need to convert it to "Wh".

To do this, use [Integration - Riemann sum integral](https://www.home-assistant.io/integrations/integration/).

**Reply by Ne0-Hack3r** [reply]
I have a similar question... I'm trying to use the AC attribute of Total Input Energy from the DP as the "Grid Consumption" value for the Energy Dashboard but need to aggregate/derive this value from 2 Delta Pro units...

I created this custom sensor:

- platform: template
  sensors:
   dp_all_ac_in:
    friendly_name: "ALL DP AC Input"
    unit_of_measurement: "Wh"
    device_class: energy
    value_template: >- 
     {{ (([ state_attr("sensor.dp1_total_input_energy","ac"), state_attr("sensor.dp2_total_input_energy","ac") ] | map('float') | sum) / 1) | round(2) }}


The value produced is what I want (an increasing measure in Wh of the total AC input energy from both units) but it does not appear in the drop down for Grid Consumption in the energy dashboard...

**Reply by vwt12eh8** [reply]
Try adding `state_class: total_increasing`. 
Without it, you won't be able to handle it in the energy dashboard.

**Reply by Ne0-Hack3r** [reply]
That is what I thought might do the trick but then I get this error when checking the configuration from Developer Tools | YAML

Invalid config for [sensor.template]: [state_class] is an invalid option for [sensor.template]. Check: sensor.template->sensors->dp_all_ac_in->state_class

So it appears the type of aggregated sensor I'm trying to create is not supposed as a custom sensor... is there another way to accomplish this?

**Reply by vwt12eh8** [reply]
Probably because you are using an old format template. 
Please refer to [this page](https://www.home-assistant.io/integrations/template/#trigger-based-template-binary-sensors-buttons-numbers-selects-and-sensors) and try writing in the new format.

**Reply by Ne0-Hack3r** [reply]
I think that did the trick. Though I'll have to wait a bit and do some AC charging to see if the data will populate into the energy dashboard... Here is what I ended up doing:

# Increasing Meter of All AC input to Delta Pros
template:
 - sensor:
   - name: "ALL DP AC Input Energy"
     unit_of_measurement: "Wh"
     device_class: energy
     state_class: total_increasing
     state: >- 
      {{ (([ state_attr("sensor.dp1_total_input_energy","ac"), state_attr("sensor.dp2_total_input_energy","ac") ] | map('float') | sum) / 1) | round(2) }}

**Reply by mattenausohz** [reply]
Is IoT also available for Delta2? Cannot find anything.

**Reply by Ne0-Hack3r** [reply]
> Is IoT also available for Delta2? Cannot find anything.

Port is not open on Delta 2 and has been closed in some recent firmware updates of some EF products... All the data/control is available for all EF products via the public MQTT server which, unfortunately, may be the only reasonable course to take for this integration... Unless we can muster enough of a loud group voice to get EF to re-open the local pots...

**Reply by robertdavies** [reply]
Ne0-Hack3r, could you share the details for the public MQTT server? I wasn't aware there was one. Thanks.

**Reply by Ne0-Hack3r** [reply]
> Ne0-Hack3r, could you share the details for the public MQTT server? I wasn't aware there was one. Thanks.

I don't have a lot of detail beyond mqtt.ecoflow.com - The HA integration would have to be able to login using your Ecoflow ID/password (setup for the app) and subscribe to the specific SN/AppKey for your unit... My understanding is that when using the app over WiFi both the app and the device communicate to that public MQTT server such that all the same data and controls that are available locally via the current integration (prior to port being closed) would be available there so, in theory, this integration would work the same but it would then be dependent on internet connectivity and the public (cloud) MQTT server (as opposed to the local one on the device itself). I suspect as well that data updates would need to be slower as updating over the internet every 3 seconds (as you can locally) would likely be unrealistic...

**Reply by robertdavies** [reply]
Thanks. I have tried connecting to the mqtt server (mqtt://mqtt.ecoflow.com:8883 (tls encryption)) using MQTT-Explorer but can't get passed username/password knock back. I have tried app username/password, serial number/password but no luck. I think they must be using an app based certificate for authentication possibly. Will have another go when more time.

**Reply by Ne0-Hack3r** [reply]
> Thanks. I have tried connecting to the mqtt server (mqtt://mqtt.ecoflow.com:8883 (tls encryption)) using MQTT-Explorer but can't get passed username/password knock back. I have tried app username/password, serial number/password but no luck. I think they must be using an app based certificate for authentication possibly. Will have another go when more time.

Let us know if you make some progress... I'm still learning about MQTT and my coding skills are a bit rusty... I really like this integration and want to preserve this type of access and logging/control of Delta Pro if at all possible.
